lib/LMSManagers/LMSNetworkManager.php

```diff
@@ -879,4 +881,86 @@ public function getFirstFreeAddress($netid)
         }
         return $ip;
     }
+
+    public function GetVlanList()
+    {
+        return $this->db->GetAll(
+            'SELECT vl.id, vlanid, description, customerid, '
+            . $this->db->Concat('cv.lastname', "' '", 'cv.name') . ' AS customername
+            FROM vlans AS vl
+            LEFT JOIN customers cv ON (vl.customerid = cv.id)
+            ORDER BY vlanid'
+        );
+    }
+
+    public function GetVlanInfo($id)
+    {
+        return $this->db->GetRow(
+            'SELECT vl.id, vlanid, description, customerid, '
+            . $this->db->Concat('cv.lastname', "' '", 'cv.name') . ' AS customername
+            FROM vlans AS vl
+            LEFT JOIN customers cv ON (vl.customerid = cv.id)
+            WHERE vl.id = ?',
+            array($id)
+        );
+    }
+
+    public function AddVlan($args)
+    {
+        $args = array(
+            'vlanid' => $args['vlanid'],
+            'description' => $args['description'],
+            'customerid' => !empty($args['customerid']) ? $args['customerid'] : null,
+        );
+
+        $result = $this->db->Execute(
+            'INSERT INTO vlans (vlanid, description, customerid) VALUES (?, ?, ?)',
+            array($args['vlanid'], $args['description'], $args['customerid'])
+        );
+
+        $args['id'] = $this->db->GetLastInsertID('vlans');
+
+        if ($args && $this->syslog) {
```

Comment:
Wiesz o tym, że wcześniejsze przypisanie $args['id'] spowoduje, że warunek testu $args będzie zawsze true?

Reply:
Generalnie test jest po to by nie dodawać do sysloga jeśli nie powiódł się INSERT.

https://github.com/lmsgit/lms/pull/1824/commits/c004593c11bb7d61fa3ba92c8c615cc3fdf8d4b1

Reply:
Wiem po co to jest, ale działałoby z błędem, bo pierwszy warunek if zawsze byłby spełniony i logowane by było nawet wówczas, gdy rekord w vlans nie został dodany.